networking/src/APIClients/interfaces/IBaseAPIClient.ts

```diff
@@ -0,0 +1,5 @@
+export enum Environment {
+  Staging = "staging",
+  Developing = "developing",
+  Testing = "testing"
```

Comment:
This is indeed janky. "Staging" is coming from the preexisting setup with the API Gateway for CreateGameSession lambda. So 'Staging' here is actually our production env. I'd like to talk with you a bit further about naming here though because one thing is hanging me up: 

I am assuming the structure for the above environments is: `developing` -> `testing/staging` -> `production`

Where `developing` is often changing through development, `testing/staging` is slightly more stable, and then `production` only changes on releases.

However, for our client side code, I think this owuld mean that our `dev` branch uses the `testing/staging` environment (so that the branch doesn't break when the backend `env` changes). It feels kind of weird to have this mismatch, where `dev` doesn't use the `developing` environment but is that as intended? 

If we could spend 5 minutes talking through this on a meeting, that would be amazing